tdms/src/arrays.cpp

```diff
@@ -27,6 +27,54 @@ void XYZVectors::set_ptr(AxialDirection d, double* ptr){
   }
 }
 
+bool XYZVectors::all_elements_less_than(double comparison_value, int vector_length,
+                                        AxialDirection component, int buffer_start) {
+  double *component_pointer;
+  switch (component) {
+    case AxialDirection::X:
+      component_pointer = x;
+      break;
+    case AxialDirection::Y:
+      component_pointer = y;
+      break;
+    case AxialDirection::Z:
+      component_pointer = z;
+      break;
+    default:
+      throw runtime_error("Error - component not recognised");
+      break;
+  }
+  for (int index = buffer_start; index < vector_length; index++) {
+    if (component_pointer[buffer_start + index] > comparison_value) { return false; }
+  }
+  return true;
+}
+bool XYZVectors::all_elements_less_than(double comparison_value, int vector_length, char component,
+                                        int buffer_start) {
+  AxialDirection d;
+  switch (component) {
+    case 'x':
+      d = AxialDirection::X;
+      break;
+    case 'y':
+      d = AxialDirection::Y;
+      break;
+    case 'z':
+      d = AxialDirection::Z;
+      break;
+    default:
+      throw runtime_error("Error - component not recognised");
+      break;
+  }
```

Comment:
Feels like we should be able to construct an `AxialDirection` with a `char`. Then this overloaded function becomes a one-liner (which could go in the header) and we only write the `switch` once.

Reply:
It's actually [already constructed](https://github.com/UCL/TDMS/blob/89fce4c9acc7e4db95d4e533dfc6e84c01a7acc0/tdms/include/globals.h#L42-L47) like this - I didn't realise!